doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+
+## Test Structure
+
+### Setup Configuration
+
+Since this feature is not related to traffic and network topology, all current topology can be applied for this test.
+
+### Ansible and Pytest
+
+No new Ansible YAML test case will be added. The test will reuse current [platform test](https://github.com/Azure/SONiC-mgmt/tree/master/tests/platform) in SONiC-mgmt. New pytest test cases will be added to [test_platform_info.py](https://github.com/Azure/SONiC-mgmt/blob/master/tests/platform/test_platform_info.py). In addition, valid_policy.json,  invalid_format_policy.json and invalid_value_policy.json will be added as thermal policy configuration file for test purpose.
+
+#### Valid policy file
+
+In the valid policy file, two policies are defined. One is for "any PSU absence", the other is for "all FAN and PSU presence".
+
+- For "any PSU absence", all FAN speed need be set to 100% and thermal control algorithm need to be disabled.
+- For "all FAN and PSU presence", thermal control algorithm need to be enabled and FAN speed should be adjusted by it.
+
+The valid_policy.json file content is like:
+
+```json
+{
+    "info_types": [
+        {
+            "type": "fan_info"
+        },
+        {
+            "type": "psu_info"
+        }
+    ],
+    "policies": [
+        {
+            "name": "any PSU absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "false"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "100"
+                }
+            ]
+        },
+        {
+            "name": "any FAN absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "false"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "100"
+                }
+            ]
+        },
+        {
+            "name": "all FAN and PSU presence",
+            "conditions": [
+                {
+                    "type": "fan.all.presence"
+                },
+                {
+                    "type": "psu.all.presence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "true"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "60"
```

Comment:
why 60? isnt the thermal control decide which will be the value based on the thermal zones?